relayer/processor/path_end_runtime.go

```diff
@@ -244,6 +264,8 @@ func (pathEnd *pathEndRuntime) MergeCacheData(ctx context.Context, cancel func()
 	pathEnd.latestHeader = d.LatestHeader
 	pathEnd.clientState = d.ClientState
 
+	pathEnd.handleCallbacks(d.IBCMessagesCache)
+
```

Comment:
`MergeCacheData` is only called sequentially by the `PathProcessor`, never simultaneously. But thanks for calling this out, because it should be private. Updated.